RELEASE_NOTES.rst

```diff
@@ -21,6 +21,555 @@
+
+  * sql_alchemy_conn
+  * sql_engine_encoding
+  * sql_engine_collation_for_ids
+  * sql_alchemy_pool_enabled
+  * sql_alchemy_pool_size
+  * sql_alchemy_max_overflow
+  * sql_alchemy_pool_recycle
+  * sql_alchemy_pool_pre_ping
+  * sql_alchemy_schema
+  * sql_alchemy_connect_args
+  * load_default_connections
+  * max_db_retries
+
+Remove requirement that custom connection UI fields be prefixed (#22607)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+  Hooks can define custom connection fields for their connection type by implementing method ``get_connection_form_widgets``.  These custom fields are presented in the web UI as additional connection attributes, but internally they are stored in the connection ``extra`` dict.  For technical reasons, previously custom field when stored in ``extra`` had to be named with a prefix ``extra__<conn type>__<field name>``.  This had the consequence of making it more cumbersome to define connections outside of the UI, since the prefix is tougher to read and work with. With #22607, we make it so that you can now define custom fields such that they can be read from and stored in ``extra`` without the prefix.
+
+  To enable this, update the dict returned by the ``get_connection_form_widgets`` method to remove the prefix from the keys.  Internally, the providers manager will still use a prefix to ensure each custom field is globally unique, but the absence of a prefix in the returned widget dict will signal to the Web UI to read and store custom fields without the prefix.  Note that this is only a change to the Web UI behavior; when updating your hook in this way, you must make sure that when your *hook* reads the ``extra`` field, it will also check for the prefixed value for backward compatibility.
+
+The webserver.X_FRAME_ENABLED configuration works according to description now (#23222).
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+  In Airflow 2.0.0 - 2.2.4 the webserver.X_FRAME_ENABLED parameter worked the opposite of its description,
+  setting the value to "true" caused "X-Frame-Options" header to "DENY" (not allowing Airflow to be used
+  in an IFrame). When you set it to "false", the header was not added, so Airflow could be embedded in an
+  IFrame. By default Airflow could not be embedded in an IFrame.
+
+  In Airflow 2.2.5 there was a bug introduced that made it impossible to disable Airflow to
+  work in IFrame. No matter what the configuration was set, it was possible to embed Airflow in an IFrame.
+
+  Airflow 2.3.0 restores the original meaning to the parameter. If you set it to "true" (default) Airflow
+  can be embedded in an IFrame (no header is added), but when you set it to "false" the header is added
+  and Airflow cannot be embedded in an IFrame.
+
+
+New Features
+^^^^^^^^^^^^
+
+- Add dynamic task mapping (`AIP-42 <https://github.com/apache/airflow/pulls?q=is%3Apr+is%3Amerged+label%3AAIP-42>`_)
```

Comment:
This link is problematic.
If in the future we will add the AIP-42 label to a PR after this release then the PR will also be displayed in this list which will confuse users.
To solve this we must use milestone as part of the link and make sure all PRs that are relevant for this release are marked with the milestone

Reply:
Good call. I set the milestone for all the PRs missing it.

```suggestion
- Add dynamic task mapping (`AIP-42 <https://github.com/apache/airflow/pulls?q=is%3Apr+is%3Amerged+label%3AAIP-42+milestone%3A%22Airflow+2.3.0%22>`_)
```